.github/workflows/platformio.yml

```diff
@@ -1,5 +1,6 @@
 on:
   push:
+  pull_request:
```

Comment:
Not strictly related to this PR but super nice to have. This way any PR will get a platform.io build you can use for testing.